pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2302,6 +2399,37 @@ func (c *Cloud) DisksAreAttached(nodeDisks map[types.NodeName][]KubernetesVolume
 	return attached, nil
 }
 
+func (c *Cloud) ResizeDisk(
+	diskName KubernetesVolumeID,
+	oldSize resource.Quantity,
+	newSize resource.Quantity) (resource.Quantity, error) {
+	awsDisk, err := newAWSDisk(c, diskName)
+	if err != nil {
+		return oldSize, err
+	}
+
+	volumeInfo, err := awsDisk.describeVolume()
+	if err != nil {
+		descErr := fmt.Errorf("AWS.ResizeDisk Error describing volume %s with %v", diskName, err)
+		return oldSize, descErr
+	}
+	requestBytes := newSize.Value()
```

Comment:
Isn’t everything resize supposed to be behind a feature gate?

Reply:
Yes but call of this function itself is beind a feature gate. The expand controller will not start unless feature gate is enabled. In addition to that, kube admin must explicitly enable `allowVolumeExpansion:true` on storage class for volume to be resizable.